_P248_HomEvap.ino

```diff
@@ -36,9 +40,11 @@ DF - Below doesn't look right; needs a RS485 to TTL(3.3v) level converter (see h
 #define P248_QUERY2         PCONFIG(4)
 #define P248_QUERY3         PCONFIG(5)
 #define P248_QUERY4         PCONFIG(6)
+#define P248_QUERY5         PCONFIG(7)
+#define P248_QUERY6         PCONFIG(8)
 #define P248_DEPIN          CONFIG_PIN3
 
-#define P248_NR_OUTPUT_VALUES   VARS_PER_TASK
+#define P248_NR_OUTPUT_VALUES   6 // VARS_PER_TASK
```

Comment:
I don't think ESPEasy will work (correctly) with 6 output values. `VARS_PER_TASK` would have been fine, just check that limit when trying to set them, or let the user select the desired subset of 4 values.

Reply:
Yep, the way it is used now, will render settings of other tasks corrupt, or maybe even crash if this is the last task.

Reply:
Ah, that explains some other weird behaviour I had earlier. But is it only possible to use 4 values then? The homevap device give much more values (even more then the 6 I have implemented for now), so how should I get those then?

Reply:
You can have multiple instances of the plugin, that would need to know about each other probably, and combined with selectable output values could create the feature you describe.
This kind of intentional multiple-instance setup has been done before, but I can't name the exact plugins, sorry.

In the extensions I did to P064 (APDS9660 Gesture sensor, available in `test` builds) I implemented 2 selectable output groups, as that sensor also has more than 4 measured values. Haven't tested (or developed) that with multiple instances in mind though, so can't tell if that would go horribly wrong. Possibly the initialization will go haywire there.

Reply:
Take a look at https://github.com/letscontrolit/ESPEasy/blob/mega/src/_P046_VentusW266.ino for multiple instance use and selectable outputs.

Reply:
Also the Eastron plugin uses something like a shared C++ instance to communicate with the device and still have several tasks to select which units of measure to output to which task.

Reply:
Ok, I'll have a look at those plugins. Is there a specific reason that only 4 values are possible? Or is it some legacy?

Reply:
It is part legacy and part just how the settings are structured.
If you look a the SettingsStruct and ExtraTaskSettings file, you'll see the arrays often have a size of `VARS_PER_TASK` (which is 4)
If you change that value, you will also be able to have more than 4 task variables, but this will render all existing settings incompatible.